qiskit/ignis/characterization/fitters.py

```diff
@@ -252,6 +253,10 @@ def _calc_data(self):
                     if self._ydata[serieslbl][-1]['std'][-1] == 0:
                         self._ydata[serieslbl][-1]['std'][-1] = 1e-4
 
+    def extract_xdata(self, result: ExperimentResult):
```

Comment:
I think this method is missing an implementation?

Reply:
This is `fitter` dependent and came about when trying to write a base `filter` class (I'm also not sold on the `filter` concept). I wrote `DiscriminationFilter` with the idea in mind that it could be the first stab of a `BaseFilter` class. I realized that I needed something like `extract_xdata` when implementing the `filter`. Here's the rational: `BaseFitter` is given the `xdata` (makes sense when you think of a T1 fit). However, this is not the case for the `IQDiscriminationFitter`: it needs to extract the xdata from the `meas_level=1` data (see function `_calc_data`). Thus, if the `filter` wants to use the discriminator's `fit` function it needs to know how to package the `xdata` so that it can be given to the `predict` function of the `fitter`. Hence the unimplemented method in `BaseFitter` which would suggest `BaseFitter` be abstract. I think a more elegant solution would be to forego the `filter` class and simply have a `predict(...)` method in `fitter` a la sklearn.